python/paddle/v2/dataset/flowers.py

```diff
@@ -0,0 +1,255 @@
+# Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+CIFAR dataset.
+
+This module will download dataset from
+http://www.robots.ox.ac.uk/~vgg/data/flowers/102/index.html 
+and parse train/test set intopaddle reader creators.
+
+This set contains images of flowers belonging to 102 different categories. 
+The images were acquired by searching the web and taking pictures. There are a
+minimum of 40 images for each category.
+
+The database was used in:
+
+Nilsback, M-E. and Zisserman, A. Automated flower classification over a large
+ number of classes.Proceedings of the Indian Conference on Computer Vision, 
+Graphics and Image Processing (2008) 
+http://www.robots.ox.ac.uk/~vgg/publications/papers/nilsback08.{pdf,ps.gz}.
+
+"""
+import cPickle
+import itertools
+from common import download
+import tarfile
+import scipy.io as scio
+from image import *
+import os
+from multiprocessing import Process
+from multiprocessing import Pool
+from multiprocessing import cpu_count
+import numpy as np
+import paddle.v2 as paddle
+__all__ = ['train', 'test', 'valid']
+
+DATA_URL = 'http://www.robots.ox.ac.uk/~vgg/data/flowers/102/102flowers.tgz'
+LABEL_URL = 'http://www.robots.ox.ac.uk/~vgg/data/flowers/102/imagelabels.mat'
+SETID_URL = 'http://www.robots.ox.ac.uk/~vgg/data/flowers/102/setid.mat'
+DATA_MD5 = '52808999861908f626f3c1f4e79d11fa'
+LABEL_MD5 = 'e0620be6f572b9609742df49c70aed4d'
+SETID_MD5 = 'a5357ecc9cb78c4bef273ce3793fc85c'
+
+
+def extract_file(tarFile):
+    '''
+    Extract tar file to tmp dir.
+    
+    Example usage:
+
+    .. code-block:: python
+        tmp = extract_file("/home/work/test.tar.gz")
+
+    :param tarFile: target tar file
+    :type tarFile: string
+    :return: extracted dir. For example: 
+            '/home/work/test/' while input is '/home/work/test.tar.gz'
+    :rtype: string
+    '''
+    base_dir = os.path.dirname(tarFile)
+    base_name = os.path.basename(tarFile)
+    if '.' in base_name:
+        base_name = base_name.split('.', 1)[0]
+    out_path = '/'.join([base_dir, base_name])
+    if not os.path.exists(out_path):
+        df = tarfile.open(tarFile, mode='r')
+        df.extractall(path=out_path)
+        df.close()
+    return out_path
+
+
+def default_mapper(sample):
+    '''
+    map image bytes data to type needed by model input layer
+    '''
+    img, label = sample
```

Comment:
This module seems reading many images from the tarball. If so, it might be great if we can call [`tarfile.next()`](https://docs.python.org/2/library/tarfile.html#tarfile.TarFile.next), which returns a TarFile objects like `tarfile.extractfile`.  But `tarfile.next()` reads files in the tarball one-by-one. This reduces the amount of disk seeks which reduces the number of moves of the magnetic head of our disk.